keps/sig-testing/1746-stage-e2e-test-framework/README.md

```diff
@@ -0,0 +1,610 @@
+- [x] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [x] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Key Terms
+
+* **e2e test framework** refers to the e2e test framework, located in
+  https://github.com/kubernetes/kubernetes/tree/master/test/e2e/framework and
+  all of its sub-packages, unless otherwise explicitly stated.
+* **Core Kubernetes** refers to code that is part of kubernetes/kubernetes,
+  including the the repository excluding the contents of the staging directory.
+  This also includes k8s.io/kubernetes/test but excludes
+  k8s.io/kubernetes/test/framework and its subdirectories (e2e test framework).
+* **Published Packages** refers to Go code packages that are available to
+  Kubernetes end-users from source code present in the Kuberentes staging
+  directory.
+
+
+## Summary
+
+<!--
+This section is incredibly important for producing high quality user-focused
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+-->
+
+In order to improve the user experience of all the people who have to maintain
+e2e tests in Kubernetes and with the aim of reducing technical debt in e2e tests
+(i.e., have more organized tests, better utility functions) we propose to work
+towards moving the e2e test framework from its current place in
+[kubernetes/test/e2e/framework](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/framework)
+to the appropriate location under `k8s.io/kubernetes/staging/src/k8s.io/e2e-framework`.
+
+Our main goal is to improve the experience of maintainers and building on top
+of what we already have instead of starting from scratch.
+
+Moving the e2e test framework into staging means that the e2e test framework
+will be used and maintained as a component independent of core Kubernetes.
+By doing this, the e2e test framework will only make use of public APIs and
+will be self-contained.
+This will help maintainers of e2e tests have tools to easily test Kubernetes
+any component that run on Kubernetes.
```

Comment:
nit: this sentence doesn't read quite right.

why do we think this will give them tools to easily test, vs existing options? what does this framework specifically offer them?

Reply:
Updated this section to try and answer this better.